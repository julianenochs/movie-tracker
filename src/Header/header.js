import React, { Component } from 'react';
import './header.scss';
import { NavLink } from 'react-router-dom';
import {
  updateIsLoggedIn,
  updateUser,
  resetMovies,
  updateFavorites,
  displayFavorites
} from '../actions';
import { connect } from 'react-redux';

class Header extends Component {
  handleSignout = () => {
    this.props.updateIsLoggedIn(false);
    this.props.updateUser('');
    this.props.updateFavorites({ favorites: [] });
    this.props.resetMovies();
  };

  render() {
    return (
      <header className='header'>
        <h1>NEWFLIX</h1>
        {!this.props.isLoggedIn && (
          <NavLink to='/login'>
            <button className='user__button'>Sign In</button>
          </NavLink>
        )}
        {!this.props.isLoggedIn && (
          <NavLink to='/register'>
            <button className='user__button'>Register</button>
          </NavLink>
        )}
        {this.props.isLoggedIn && (
          <NavLink to='/'>
            <button onClick={this.handleSignout} className='user__button'>
              Sign Out
            </button>
          </NavLink>
        )}
        {this.props.user.email !== '' && <h3>{this.props.user.email}</h3>}
        {this.props.isLoggedIn && (
          <NavLink to='/favorites'>
            <button className='user__button'>Favorites</button>
          </NavLink>
        )}
      </header>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  updateIsLoggedIn: boolean => dispatch(updateIsLoggedIn(boolean)),
  updateUser: email => dispatch(updateUser(email)),
  updateFavorites: favorites => dispatch(updateFavorites(favorites)),
  resetMovies: () => dispatch(resetMovies()),
  displayFavorites: favoriteMovies => dispatch(displayFavorites(favoriteMovies))
});

const mapStateToProps = state => ({
  user: state.user,
  isLoggedIn: state.isLoggedIn,
  movies: state.movies
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
