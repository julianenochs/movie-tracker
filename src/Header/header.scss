$header-color: #B27A27;
$shadow-color-one: #6D4B18;
$shadow-color-two: #4C3410;
$button-color: #004263;
$button-shadow: #003149;


h1 {
  font-family: 'Fascinate Inline', cursive;
  text-shadow: 0 1px 0px $shadow-color-one,
    1px 0 0px $shadow-color-two,
    1px 2px 1px $shadow-color-one,
    2px 1px 1px $shadow-color-two,
    2px 3px 2px $shadow-color-one,
    3px 2px 2px $shadow-color-two,
    3px 4px 2px $shadow-color-one;
  }
  
  .header {
  align-items: center;
  color: $header-color;
  display: flex;
  font-family: 'Fascinate Inline', cursive;
  font-size: 2em;
  height: 10vh;
  letter-spacing: 7px;
  justify-content: space-between;
  width: 85%;
}

.user__button {
  background-color: $button-color;
  border: none;
  border-radius: 5px;
  box-shadow: 4px 4px $button-shadow;
  color: white;
  font-family: 'Barlow Condensed', sans-serif;
  font-size: .7em;
  height: 40px;
  letter-spacing: 5px;
  width: 120px;
}

h3 {
  font-family: 'Staatliches', cursive;

}