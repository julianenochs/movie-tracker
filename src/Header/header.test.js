import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';

describe('Header', () => {
  let wrapper;
  const mockMovies = [
    {
      popularity: 471.315,
      vote_count: 4211,
      video: false,
      poster_path: '/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg',
      id: 475557,
      adult: false,
      backdrop_path: '/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg',
      original_language: 'en',
      original_title: 'Joker',
      title: 'Joker',
      vote_average: 8.6,
      overview:
        'During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.',
      release_date: '2019-10-04',
      isFavorite: false
    }
  ];

  const mockUpdateIsLoggedIn = jest.fn();
  const mockUpdateUser = jest.fn();
  const mockUpdateFavorites = jest.fn();
  const mockResetMovies = jest.fn();
  const mockDisplayFavorites = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Header
        user={'Peerat Test'}
        isLoggedIn={true}
        movies={mockMovies}
        updateIsLoggedIn={mockUpdateIsLoggedIn}
        updateUser={mockUpdateUser}
        updateFavorites={mockUpdateFavorites}
        resetMovies={mockResetMovies}
        displayFavorites={mockDisplayFavorites}
      />
    );
  });

  it('should match the LoginForm Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
