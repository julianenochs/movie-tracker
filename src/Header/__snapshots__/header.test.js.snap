// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the LoginForm Snapshot 1`] = `ShallowWrapper {}`;
